lib/rate_map.c

```diff
@@ -153,25 +157,24 @@ rate_map_position_to_mass(rate_map_t *self, double pos)
 {
     const double *position = self->position;
     const double *rate = self->rate;
-    double offset;
+    double base, offset;
     size_t index;
 
     assert(pos >= 0.0);
-    if (pos <= position[0]) {
+    if (pos <= 0.0) {
```

Comment:
Is there much value in the assert here now if we're explicitly handling the < 0 case?

Reply:
With #1203 merged I would say yes it is of value. But I might be too 'assert happy' and 'assertive' :rofl: .
I'll just take it out pre #1203 to make merging #1203 easier.

The idea with a traditional assert.h assert() here is that if this condition is getting violated, even by the slightest rounding error, it's worth having debug builds crash to investigate, even though a release build will handle it in a reasonable and well defined way.

Reply:
Crap, I just deleted my previous comment. Anyway, I was basically saying the same thing as the other assert. I'll remove it for this pre #1203 code.